### PR TITLE
Lps 87184 - Previewing members of a segment on a modal

### PR DESCRIPTION
**Pull request by andresfulla**
@epgarcia @thektan

### CODE REVIEW

**Comment by epgarcia** [comment]
Hi @bryceosterhaus,

If you could please prioritize this one it'd be awesome. We'll have to work on top of this changes in several tasks, and it'd be better if we consolidate them first.

Thanks!!

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/shared/ClayButton.es.js [hunk 1]
```diff
@@ -52,6 +58,7 @@ class ClayButton extends Component {
 		const classes = getCN(
 			'btn',
 			`${stylePrefix}${style}`,
+			size ? `btn-${size}` : null,
```
Comment (bryceosterhaus) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/shared/ClayButton.es.js]: This should be in the object below as `["btn-${size}"]: size`

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js [hunk 2]
```diff
@@ -119,7 +124,8 @@ class ContributorBuilder extends React.Component {
 				return {
 					contributors
 				};
-			}
+			},
+			() => onQueryChange && onQueryChange()
```
Comment (bryceosterhaus) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js]: Would it make sense to default `onQueryChange` to a no-op? That way you can pass it by itself as the second argument.

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js [hunk 3]
```diff
@@ -67,7 +69,8 @@ class ContributorBuilder extends React.Component {
 				}
 
 				return diffState;
-			}
+			},
+			() => onQueryChange && onQueryChange()
```
Comment (bryceosterhaus) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js]: see below

**Comment by epgarcia** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by epgarcia** [comment]
Hi @bryceosterhaus ,

Sorry for that, the PR should have been sent rebased with the latest changes. I'll ask @andresfulla to rebase it, apply your suggestions and resend it on the same branch.

Thanks for the prompt response

**Comment by andresfulla** [comment]
@bryceosterhaus Thanks! Just rebased and fixed those callbacks and classname.

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js [hunk 4]
```diff
@@ -10,6 +10,10 @@ import {CONJUNCTIONS} from '../../utils/constants.es';
 import Conjunction from './Conjunction.es';
 
 class ContributorBuilder extends React.Component {
+	static propTypes = {
```
Comment (bryceosterhaus) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js]: Shouldn't this be `defaultProps`?

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js [hunk 5]
```diff
@@ -10,6 +10,10 @@ import {CONJUNCTIONS} from '../../utils/constants.es';
...
+		onQueryChange: null
```
Comment (thektan) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js]: Should this be defaulted to an empty function instead?
```javascript
onQueryChange: () => {}
```

**Comment by thektan** [comment]
@bryceosterhaus @andresfulla Opened a new pr with the `defaultProps` change: https://github.com/bryceosterhaus/liferay-portal/pull/22

I also moved the `propTypes` definitions inside the class to the component structures consistent.

bcae50a...3449a5e

**Comment by bryceosterhaus** [comment]
https://github.com/brianchandotcom/liferay-portal/pull/67100

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js (hunk 5 above)
Comment (andresfulla) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js]: it was late 🤦‍♂️

**Pull request closed without merging**